lib/rules/no-restricted-imports.js

```diff
@@ -95,7 +104,16 @@ module.exports = {
             return {};
         }
 
-        const ig = ignore().add(restrictedPatterns);
+        const restrictedPatternsMatcher = ignore().add(restrictedPatterns);
+
+        /**
+         * Checks to see if '*' is being used to import everything.
+         * @param {node} node representing the restricted path reference
+         * @returns {boolean} whether everything is imported or not
+         */
+        function isEverythingImported(node) {
+            return !!(node && node.parent && node.parent.tokens && node.parent.tokens.find(token => token.value === "*"));
```

Comment:
Instead of looking for a `*` token, would it be possible to check whether the parent node has type `ImportNamespaceSpecifier`?

Reply:
Let me play around with that.  I agree the `*` token is not my first choice, but when I was originally writing this I wasn't finding a differentiator like that.  I will love it if this works.

Reply:
Worked perfectly.  I have updated this as my fourth commit.